### PR TITLE
#18673 : Adding validation to the objects that have a high chance of …

### PR DESCRIPTION
**Pull request by jcastro-dotcms**
…causing NPEs because of incorrect/unexpected content information during the reindex. Improving error logging. These errors are very likely because the customer who experienced them comes from a very old version of dotCMS.

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESMappingAPIImpl.java [hunk 1]
```diff
@@ -189,10 +189,36 @@ public String toJson(final Contentlet contentlet) throws DotMappingException {
             loadRelationshipFields(contentlet, contentletMap, sw);
 
 			final Identifier contentIdentifier = APILocator.getIdentifierAPI().find(contentlet);
+            if (null == contentIdentifier || !UtilMethods.isSet(contentIdentifier.getId())) {
+                final String errorMsg = String.format("Identifier '%s' was not found via API.", contentlet
+                        .getIdentifier());
+                throw new Exception(errorMsg);
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESMappingAPIImpl.java:195]: Codacy Issue found: [Avoid throwing raw exception types.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5707604)

Comment (dsilvam) [inline dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESMappingAPIImpl.java:195]: DotDataException

**Review by dsilvam — APPROVED** [review]

**Review by erickgonzalez — APPROVED** [review]

File: dotCMS/src/integration-test/java/com/dotcms/content/elasticsearch/business/ESMappingAPITest.java [hunk 2]
```diff
@@ -143,6 +145,106 @@ public void test_toMap_fileasset_txt_shouldSuccess() throws Exception {
...
+     * @throws DotDataException
+     */
+    @Test(expected = DotMappingException.class)
+    public void whenContentletHasNotContentletVersionInfo() throws DotDataException {
+        final ESMappingAPIImpl esMappingAPI    = new ESMappingAPIImpl();
+
+        final ContentType contentType = new ContentTypeDataGen().nextPersisted();
+        final Contentlet contentlet = new ContentletDataGen(contentType.id()).nextPersisted();
+
+        final ContentletVersionInfo versionInfo = APILocator.getVersionableAPI()
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/content/elasticsearch/business/ESMappingAPITest.java:241]: Codacy found an issue: [Avoid unused local variables such as 'versionInfo'.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5707604)

File: dotCMS/src/integration-test/java/com/dotcms/content/elasticsearch/business/ESMappingAPITest.java [hunk 3]
```diff
@@ -26,24 +27,24 @@
...
 import com.dotcms.datagen.FieldDataGen;
 import com.dotcms.datagen.FileAssetDataGen;
 import com.dotcms.datagen.SiteDataGen;
 import com.dotcms.util.CollectionsUtils;
 import com.dotcms.util.IntegrationTestInitService;
 import com.dotmarketing.beans.Host;
-import com.dotmarketing.business.APILocator;
-import com.dotmarketing.business.RelationshipAPI;
-import com.dotmarketing.business.UserAPI;
+import com.dotmarketing.business.*;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/content/elasticsearch/business/ESMappingAPITest.java:40]: Codacy found an issue: [Avoid unused imports such as 'com.dotmarketing.business'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5707604)

**Review by jdotcms — APPROVED** [review]

**Pull request merged**